app/models/concerns/registrations/persistence.rb

```diff
@@ -0,0 +1,60 @@
+module Registrations::Persistence
+  extend ActiveSupport::Concern
+
+  def save
+    return false unless valid?
+
+    User.transaction do
```

Comment:
Le pondría:
```suggestion
    ActiveRecord::Base.transaction do
```

Aunque quede más feo, porque no es "un solo modelo" el que se toca ahí.